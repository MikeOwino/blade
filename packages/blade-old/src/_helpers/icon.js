// import icons from '../icons';

// todo remove this
const getIconNames = () => Object.keys({});

export { getIconNames };
