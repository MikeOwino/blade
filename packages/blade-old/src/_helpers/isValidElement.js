import { isValidElementType } from 'react-is';

const isValidElement = (element) => {
  if (typeof element === 'string' || typeof element === 'number') return false;
  return isValidElementType(element);
};

export default isValidElement;
