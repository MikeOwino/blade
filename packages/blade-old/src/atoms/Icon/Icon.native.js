import React from 'react';
import PropTypes from 'prop-types';
import { useTheme } from 'styled-components/native';
import Svg from 'react-native-svg';
import { getColorKeys, getColor } from '../../_helpers/theme';
import automation from '../../_helpers/automation-attributes.native';

const iconSize = {
  xsmall: 12,
  small: 16,
  medium: 20,
  large: 24,
  xlarge: 28,
};

const Icon = ({ size, fill, testID, children, ...rest }) => {
  const theme = useTheme();

  return (
    <Svg
      accessibilityElementsHidden={true}
      fill={getColor(theme, fill)}
      height={iconSize[size]}
      width={iconSize[size]}
      {...automation(testID)}
      {...rest}
    >
      {children}
    </Svg>
  );
};

export const IconPropTypes = {
  size: PropTypes.oneOf(['xsmall', 'small', 'medium', 'large', 'xlarge']),
  fill: PropTypes.oneOf(getColorKeys()),
  testID: PropTypes.string,
  children: PropTypes.node,
};

Icon.propTypes = IconPropTypes;
Icon.defaultProps = {
  size: 'medium',
  testID: 'ds-icon',
  fill: 'shade.950',
};

export default Icon;
