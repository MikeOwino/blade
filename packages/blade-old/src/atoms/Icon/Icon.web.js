import React from 'react';
import PropTypes from 'prop-types';
import { useTheme } from 'styled-components';
import automation from '../../_helpers/automation-attributes.web';
import { getColorKeys, getColor } from '../../_helpers/theme';

const iconSize = {
  xsmall: 12,
  small: 16,
  medium: 20,
  large: 24,
  xlarge: 28,
};

const Icon = ({ size, fill, children, ...rest }) => {
  const theme = useTheme();

  return (
    <svg
      focusable={false}
      aria-hidden={true}
      fill={getColor(theme, fill)}
      height={iconSize[size]}
      width={iconSize[size]}
      {...automation('ds-icon')}
      {...rest}
    >
      {children}
    </svg>
  );
};

export const IconPropTypes = {
  size: PropTypes.oneOf(['xsmall', 'small', 'medium', 'large', 'xlarge']),
  fill: PropTypes.oneOf(getColorKeys()),
  children: PropTypes.node,
};

Icon.propTypes = IconPropTypes;
Icon.defaultProps = {
  size: 'medium',
  fill: 'sapphire.800',
};

export default Icon;
