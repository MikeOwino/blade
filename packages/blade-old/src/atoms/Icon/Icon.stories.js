import React from 'react';
import { storiesOf } from '@storybook/react';
import { select } from '@storybook/addon-knobs';
import { getColorKeys } from '../../_helpers/theme';
import * as icons from '../../icons';
import Icon from './Icon';

const sizes = {
  xsmall: 'xsmall',
  small: 'small',
  medium: 'medium',
  large: 'large',
  xlarge: 'xlarge',
};

storiesOf('Icon', module)
  .addParameters({
    component: Icon,
  })
  .add('default', () => {
    return (
      <>
        {Object.values(icons).map((Ico, index) => {
          return (
            <Ico
              key={index}
              fill={select('fill', getColorKeys(), 'purple.900')}
              size={select('sizes', sizes, 'medium')}
              data-testid="icon"
            />
          );
        })}
      </>
    );
  });
