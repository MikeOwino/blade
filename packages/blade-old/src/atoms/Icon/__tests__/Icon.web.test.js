import React from 'react';
import { renderWithTheme } from '../../../_helpers/testing';
import Icon from '../index';

const svgPath = (
  <path
    fillRule="evenodd"
    clipRule="evenodd"
    d="M0 6a3 3 0 013-3h18a3 3 0 013 3v12a3 3 0 01-3 3H3a3 3 0 01-3-3V6zm2 0a1 1 0 011-1h18a1 1 0 011 1v3H2V6zm20 5v7a1 1 0 01-1 1H3a1 1 0 01-1-1v-7h20z"
  />
);

describe('<Icon />', () => {
  describe('size', () => {
    it('should render a medium icon', () => {
      const { container } = renderWithTheme(<Icon size="medium">{svgPath}</Icon>);
      expect(container).toMatchSnapshot();
    });
    it('should render a small icon', () => {
      const { container } = renderWithTheme(<Icon size="small">{svgPath}</Icon>);
      expect(container).toMatchSnapshot();
    });
  });
  describe('fill', () => {
    it('should render a color filled icon', () => {
      const { container } = renderWithTheme(<Icon fill="shade.980">{svgPath}</Icon>);
      expect(container).toMatchSnapshot();
    });
  });
});
