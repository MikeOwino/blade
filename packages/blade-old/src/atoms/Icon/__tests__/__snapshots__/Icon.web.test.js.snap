// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> fill should render a color filled icon 1`] = `
<div>
  <svg
    aria-hidden="true"
    data-testid="ds-icon"
    fill="rgba(22, 47, 86, 0.87)"
    focusable="false"
    height="20"
    width="20"
  >
    <path
      clip-rule="evenodd"
      d="M0 6a3 3 0 013-3h18a3 3 0 013 3v12a3 3 0 01-3 3H3a3 3 0 01-3-3V6zm2 0a1 1 0 011-1h18a1 1 0 011 1v3H2V6zm20 5v7a1 1 0 01-1 1H3a1 1 0 01-1-1v-7h20z"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;

exports[`<Icon /> size should render a medium icon 1`] = `
<div>
  <svg
    aria-hidden="true"
    data-testid="ds-icon"
    fill="rgba(43, 131, 234, 1.0 )"
    focusable="false"
    height="20"
    width="20"
  >
    <path
      clip-rule="evenodd"
      d="M0 6a3 3 0 013-3h18a3 3 0 013 3v12a3 3 0 01-3 3H3a3 3 0 01-3-3V6zm2 0a1 1 0 011-1h18a1 1 0 011 1v3H2V6zm20 5v7a1 1 0 01-1 1H3a1 1 0 01-1-1v-7h20z"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;

exports[`<Icon /> size should render a small icon 1`] = `
<div>
  <svg
    aria-hidden="true"
    data-testid="ds-icon"
    fill="rgba(43, 131, 234, 1.0 )"
    focusable="false"
    height="16"
    width="16"
  >
    <path
      clip-rule="evenodd"
      d="M0 6a3 3 0 013-3h18a3 3 0 013 3v12a3 3 0 01-3 3H3a3 3 0 01-3-3V6zm2 0a1 1 0 011-1h18a1 1 0 011 1v3H2V6zm20 5v7a1 1 0 01-1 1H3a1 1 0 01-1-1v-7h20z"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;
